### PR TITLE
Feature/add refresh token interceptor

### PR DESCRIPTION
**Pull request by nunommts**
**What's this do?**
This PR adds a OKHttp interceptor to handle a token expired event. When the interceptor gets a 401 response, it will check for three fields set by the client: username, password and an auth url to where it should create a new request in order to refresh the token. If one of these fields is not there or doesn't have a valid value, the interceptor will ignore and proceed with the original request. If all the fields are there and are valid, then a new request will be performed to refresh the token and the original request will be retried with the new access token. This original request retry's response will include a header with the new access token that should be handled by the client.

**Why are we doing this? (w/ JIRA link if applicable)**
There's a [new auth method to be integrated in the Palace app](https://ebce-lyrasis.atlassian.net/browse/PP-90) that will include a "refresh token" feature and adding an interceptor in this module is the easiest way to cover all the existing requests.

**How should this be tested? / Do these changes have associated tests?**
Follow the reproducing steps of [this PR](https://github.com/ThePalaceProject/android-core/pull/205)

**Dependencies for merging? Releasing to production?**
[This PR](https://github.com/ThePalaceProject/android-core/pull/205) needs to be merged too.

**Have you updated the changelog?**
Yes

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
Tested by @nunommts

### CODE REVIEW

File: org.librarysimplified.http.refresh_token/src/main/java/org/librarysimplified/http/refresh_token/internal/LSSimplifiedRefreshTokenJSON.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper
+import java.io.InputStream
+
+/**
+ * Functions to deserialize an access token.
+ */
+
+object LSSimplifiedRefreshTokenJSON {
+  fun deserializeFromStream(
+    stream: InputStream?
```
Comment (io7m) [inline org.librarysimplified.http.refresh_token/src/main/java/org/librarysimplified/http/refresh_token/internal/LSSimplifiedRefreshTokenJSON.kt:12]: Why is this nullable?

**Comment by io7m** [comment]
Looks like the platform submodule needs updating.

File: org.librarysimplified.http.refresh_token/src/main/java/org/librarysimplified/http/refresh_token/internal/LSSimplifiedRefreshTokenJSON.kt (hunk 1 above)
Comment (nunommts) [reply org.librarysimplified.http.refresh_token/src/main/java/org/librarysimplified/http/refresh_token/internal/LSSimplifiedRefreshTokenJSON.kt:12]: I was passing the inputstream directly from the response to there, but I guess I can check its nullability before that!

**Review by io7m — APPROVED** [review]
Looks good!

**Review by io7m — APPROVED** [review]
Still looks good to me!

**Pull request merged**